app/services/order_details/reconciler.rb

```diff
@@ -44,12 +53,20 @@ def to_be_reconciled
       @params.select { |_order_detail_id, params| params[:reconciled] == "1" }
     end
 
-    def reconcile(order_detail, params)
-      order_detail.reconciled_at = @reconciled_at
-      order_detail.assign_attributes(allowed(params))
-      order_detail.reconciled_note = @bulk_reconcile_note if @bulk_reconcile_note.present?
-      order_detail.deposit_number = @bulk_deposit_number if @bulk_deposit_number.present?
-      order_detail.change_status!(OrderStatus.reconciled)
+    def update_status(order_detail, params)
+      if @order_status == "reconciled"
+        order_detail.reconciled_at = @reconciled_at
+        order_detail.assign_attributes(allowed(params))
```

Comment:
nit pick - I think this line runs in both branches, so probably you could do this first before checking `@order_status`